tasm-lib/src/data_type.rs

```diff
@@ -50,6 +50,24 @@ impl Display for StructType {
 }
 
 impl DataType {
+    /// See [`BFieldCodec::static_length`].
+    pub(crate) fn static_length(&self) -> Option<usize> {
+        match self {
+            DataType::Bool => bool::static_length(),
+            DataType::U32 => u32::static_length(),
+            DataType::U64 => u64::static_length(),
+            DataType::U128 => u128::static_length(),
+            DataType::Bfe => BFieldElement::static_length(),
+            DataType::Xfe => XFieldElement::static_length(),
+            DataType::Digest => Digest::static_length(),
+            DataType::List(_) => None,
+            DataType::Array(a) => Some(a.length * a.element_type.static_length()?),
+            DataType::Tuple(t) => t.iter().map(|dt| dt.static_length()).sum(),
+            DataType::VoidPointer => None,
+            DataType::StructRef(s) => s.fields.iter().map(|(_, dt)| dt.static_length()).sum(),
+        }
+    }
+
```

Comment:
I'm surprised this method doesn't already exists! But if it doesn't (I couldn't find it) we should definitely have it!